src/imfill.jl

```diff
@@ -0,0 +1,53 @@
+"""
+ Image filling Algorithm
+
+ ```
+ filled_img = imfill(img, min_size)
+ filled_img = imfill(img, min_size, threshold)
+ filled_img = imfill(img, min_size, threshold, value)
+ filled_img = imfill(img, min_size, threshold, value, connectivity)
+ ```
+
+ Connected components of an image is found using flood-fill algorithm and returns a copy of
+ the original image after filling objects that are smaller than the specified min_size.
+
+
+ Parameters:
+
+  -  img            = Input image (Float valued)
```

Comment:
So this algorithm is supposed to work on binary images, correct? Shouldn't we make the input image already a Bool array?

Reply:
Yeah, the algorithm  works on binary images, but if we take the input image as Bool array then we won't be able to utilize the `threshold` parameter which is used to give more flexibility to fill objects.

Reply:
I guess more flexibility could be achieved by letting the user threshold the image outside the function before filling it. The threshold method you are using is simple to implement, and we could foresee other more complicated methods in the real world.

Reply:
Also, I propose to input the image in the function in the same format as it has been loaded i.e in it's `colorview`, RGB, Gray etc. . This will provide more ease to use the function. What's your opinion?

Reply:
Agreed!! Should I add the `descriptor` parameter or a function to access connected components before filling?

Reply:
I plan to use https://github.com/JuliaImages/Images.jl/blob/96e10ad90211222dbb95afe3e0f523b6139b8eca/src/connected.jl#L247 and some other functions there to help user identify the `min_size` they should provide to fill objects i.e to reduce trial-errors and increase ease of use.

Reply:
ping @juliohm

Reply:
Any other things before this can be merged @juliohm ?

Reply:
I've added some comments below :+1: